### DISCUSSION TITLE
Tuya IoT Platform caricati gli elementi ma non gestibili da HA

### DISCUSSION THREAD

**Discussion by nonsviluppo** [indomus/forum#1452] [HUB: Home Assistant]
Buona sera a tutti, 
scusate il disturbo volevo chiedere se qualcuno sa come aiutarmi, il fatto è che da quando ho installato Tuya IoT Platform sustituendo la vecchia non riesco più a gestire i miei elementi . se utilizzo l'App Smart Life va tutto bene e addirittura li vedo accendersi anche su home assistant , mentre se li voglio gestire da HA non riesco non mi fa cliccare sui comandi , inoltre anche la misurazione della temperatura è errata. qualcuno sa come risolvere?
allego un'immagine per farvi capire meglio.
![termostato](https://user-images.githubusercontent.com/58274687/152549656-de45c20a-a324-4250-b2fc-35a5274538ce.jpg)
vi ringrazio per il tempo e la pazienza che potete dedicarmi 
ciao a tutti

**Comment by Franco2255** [comment]
Ciao stesso problema ho segnalato ila problema a TUYA e questa è la loro risposta.
lTuya engineer02-03 23:12:46
Hi dear, sorry to keep you waiting. It will take more time since the issue is being technically identified. We will contact you in time for progress, and you will be notified on the E-mail. Thank you for your patience and understanding~

**Reply by Newreader** [reply]
Puoi pubblicare i log di riferimento?
Hai HA installato su docker? 
È un problema che riguarda il rinnovo del Token di accesso al cloud?

**Reply by Franco2255** [reply]
il problema è la loro configurazione su Tuya se hai letto an altro post si puo risolvere ma al primo riavvio di HA ritorna come prima.
Questa è la loro configurazione sul cloud:
Standard Instruction Set
<html>
<body>


child_lock | Boolean | "{true,false}"
-- | -- | --
temp_set | Integer | {   "unit": "℃",   "min": 10,   "max": 70,   "scale": 1,   "step": 5 }
eco | Boolean | "{true,false}"
upper_temp | Integer | {   "unit": "℃",   "min": 0,   "max": 100,   "scale": 0,   "step": 5 }


</body>
</html>

JSON_EDITOR
[
  {
    "code": "switch",
    "value": true
  },
  {
    "code": "child_lock",
    "value": false
  },
  {
    "code": "temp_set",
    "value": 44
  },
  {
    "code": "eco",
    "value": false
  },
  {
    "code": "upper_temp",
    "value": 44
  }
]

**Reply by Newreader** [reply]
Scusami ma non hai risposto alle mie domande.
Comunque il problema, se ho ben capito, si risole aggiornando a mano il file "tuya_iot/openapi.py":
https://github.com/tuya/tuya-iot-python-sdk/pull/51/files
Fammi sapere, se vuoi.

**Reply by Franco2255** [reply]
ho provato ma no si riesce lo puoi fare da HA ma al primo riavvio di HA ritorna come leggi questo post
Problema Termostato Tuya Imported #1379

**Reply by Newreader** [reply]
Hai un riferimento alla problematica nei log che puoi pubblicare?
Quale versione di HA hai? Dalla versione 2021.12.6 sono state risolte diverse problematiche relative a Tuya ivi comprese alcune riguardanti il clima.

**Reply by Franco2255** [reply]
la mia versione 2022.01.1 in Versione Docker19.03.11   
Sistema operativo Ubuntu 20.04 LTS

**Reply by Franco2255** [reply]
il problema lo possono risolvere solo loro per adesso.

**Comment by nonsviluppo** [comment]
ciao Newreader,
scusa il ritardo, ma i log di riferimento come si pubblicano? non so se intendevi a me o a franco2255

**Reply by Newreader** [reply]
Non ti preoccupare. Leggi l'ultima risposta di Franco2255.

**Reply by Franco2255** [reply]
Buongiorno questa è l'ultima risposta degli ingegneri di TUYA:
Tuya engineer02-08 04:10:32
Hello, you just need to edit the code on HomeAssistant so that 4.4 is displayed as 22 (4.4*5=22.0). You can submit an issue in the official github_HA to obtain the operation process of editing the code.
https://github.com/home-assistant/core/issues

**Comment by nonsviluppo** [comment]
Grazie di cuore per le info, cmq da quello che capisco non si può risolvere al momento, ma avendo gli indirizzi ip dei due termostati che poi sono due termostati decdeal non posso usare qualche altro strumento per controllarli con HA ?
ancora grazie
